jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/quartz/DeploymnetsAwareStdJDBCDelegate.java

```diff
@@ -0,0 +1,148 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.core.timer.impl.quartz;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.quartz.impl.jdbcjobstore.StdJDBCDelegate;
+import org.quartz.utils.Key;
+import org.slf4j.Logger;
+
+
+public class DeploymnetsAwareStdJDBCDelegate extends StdJDBCDelegate {
+
+    private QuertzUtils quertzUtils = new QuertzUtils();
+    
+    public DeploymnetsAwareStdJDBCDelegate(Logger logger, String tablePrefix, String instanceId, Boolean useProperties) {
+        super(logger, tablePrefix, instanceId, useProperties);
+    }
+
+    public DeploymnetsAwareStdJDBCDelegate(Logger logger, String tablePrefix, String instanceId) {
+        super(logger, tablePrefix, instanceId);
+    }
+    
+    @Override
+    public List selectTriggerToAcquire(Connection conn, long noLaterThan, long noEarlierThan)
+        throws SQLException {
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        List nextTriggers = new LinkedList();
+        try {
+            List<String> deploymentIds = quertzUtils.getDeployments();
+            ps = conn.prepareStatement(rtp(quertzUtils.nextTriggerQuery(deploymentIds)));
+            
+            // Try to give jdbc driver a hint to hopefully not pull over 
+            // more than the few rows we actually need.
+            ps.setFetchSize(5);
+            ps.setMaxRows(5);
+            
+            ps.setString(1, STATE_WAITING);
+            ps.setBigDecimal(2, new BigDecimal(String.valueOf(noLaterThan)));
+            ps.setBigDecimal(3, new BigDecimal(String.valueOf(noEarlierThan)));
+            int index = 4;
+            for (String deployment : deploymentIds) {
+                ps.setString(index++, deployment);
+            }
+            
+            rs = ps.executeQuery();
+            
+            while (rs.next() && nextTriggers.size() < 5) {
+                nextTriggers.add(new Key(
+                        rs.getString(COL_TRIGGER_NAME),
+                        rs.getString(COL_TRIGGER_GROUP)));
+            }
+            
+            return nextTriggers;
+        } finally {
+            closeResultSet(rs);
+            closeStatement(ps);
+        }      
+    }
+    
+    @Override
+    public int countMisfiredTriggersInStates(
+                                             Connection conn, String state1, String state2, long ts) throws SQLException {
```

Comment:
This is some weird formatting style :) Was it done automatically by KIE code style? If so, then we have a bug in it.

Reply:
will double check